src/Version.php

```diff
@@ -321,6 +332,72 @@ public function prefix($prefix = 'v')
         return $prefix . $this->toString();
     }
 
+    /**
+     * Tag Git repo with version
+     *
+     * @author Joshua Young (jny986)
+     * @param $gitPath
+     * @param bool $prefix
+     * @return bool
+     * @throws \Cz\Git\GitException
+     */
+    public function gitTag($gitPath = false, $prefix = false)
+    {
+        $gitPath ?: $gitPath = __DIR__ . '/../../../../..';
+        if (!isset($this->repo) || empty($this->repo)) {
+            $this->initGitRepo($gitPath);
+        }
+        $version = $this->prefix($prefix ?: '');
+        if (!in_array($version, $this->gitTags($gitPath))) {
+            $result = $this->repo->createTag($version);
+            if ($result instanceof Exception) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Get all tags for the Git repo
+     *
+     * @author Joshua Young (jny986)
+     * @param bool $gitPath
+     * @return bool
+     * @throws \Cz\Git\GitException
+     */
+    public function gitTags($gitPath = false)
+    {
+        if (!isset($this->repo) || empty($this->repo)) {
+            $this->initGitRepo($gitPath);
+        }
+        $result = $this->repo->getTags();
+        if ($result instanceof Exception) {
+            return false;
+        }
+        return $result;
+    }
+
+    /**
+     * Remove Git tag from repository
+     *
+     * @author Joshua Young (jny986)
+     * @param bool $gitPath
+     * @param bool $prefix
+     * @return bool
+     * @throws \Cz\Git\GitException
+     */
+    public function gitTagRemove($gitPath = false, $prefix = false)
+    {
+        if (!isset($this->repo) || empty($this->repo)) {
+            $this->initGitRepo($gitPath);
+        }
+        $result = $this->repo->removeTag($this->prefix($prefix ?: ''));
+        if ($result instanceof Exception) {
+            return false;
+        }
+        return true;
+    }
+
```

Comment:
I don't think we should have the ability to list or remove tags through this extension. If someone needs those abilities they should look into a more fully-fledged git library instead of using this library.